google/cloud/storage/internal/hash_validator.cc

```diff
@@ -25,13 +25,61 @@ namespace storage {
 inline namespace STORAGE_CLIENT_NS {
 namespace internal {
 
+void HashValidator::CheckResult(std::string const& msg,
```

Comment:
As a way to better test this in the case where we don't have exceptions, could we make this return a boolean, or a optional<HashMismatchError>, a pair of the two, etc.?

Reply:
Hmmm, then you have two ways to signal errors, which is always sort of inconvenient.  I guess this should be a private function really, the only reason it is exposed is to make the unit tests easier to write.  Should I make it private?  It is in the `internal` namespace anyway, so should not be used.